websites/R/RTLplay/presence.ts

```diff
@@ -0,0 +1,555 @@
+				presence.getSetting<boolean>("useChannelName"),
+				presence.getSetting<boolean>("privacy"),
+				presence.getSetting<boolean>("timestamp"),
+				presence.getSetting<number>("buttons"),
+				presence.getSetting<boolean>("usePosterImage"),
+			]);
+
+	if (oldLang !== lang) {
+		oldLang = lang;
+		strings = await getStrings();
+	}
+
+	switch (true) {
+		/* MAIN PAGE (Page principale)
+
+		(https://www.rtlplay.be/) */
+		case pathname === "/" ||
+			(pathname.split("/")[1] === "rtlplay" &&
+				pathname.split("/").length <= 2): {
+			presenceData.details = (await strings).browsing;
+
+			presenceData.smallImageKey = Assets.Reading;
+			presenceData.smallImageText = (await strings).browsing;
+
+			if (!privacy) {
+				presenceData.state = (await strings).viewHome;
+
+				if (time) presenceData.startTimestamp = browsingTimestamp;
+			}
+			break;
+		}
+
+		/* RESEARCH PAGE (Page de recherche)
+
+		(https://www.rtlplay.be/rtlplay/recherche) */
+		case pathname.split("/")[2] === "recherche": {
+			if (privacy) presenceData.details = (await strings).searchSomething;
+			else {
+				presenceData.details = JSON.parse(
+					document
+						.querySelector("ol.search__results")
+						?.getAttribute("data-tracking")
+				).searchQuery
+					? (await strings).search
+					: (await strings).searchSomething;
+				presenceData.state = JSON.parse(
+					document
+						.querySelector("ol.search__results")
+						?.getAttribute("data-tracking")
+				).searchQuery.term;
+
+				if (time) presenceData.startTimestamp = browsingTimestamp;
+
+				if (buttons) {
+					presenceData.buttons = [
+						{
+							label: (await strings).buttonViewPage, // Need to be a general string
+							url: href, // We are not redirecting directly to the raw video stream, it's only the media page
+						},
+					];
+				}
+			}
+
+			presenceData.smallImageKey = Assets.Search;
+			presenceData.smallImageText = (await strings).search;
+			break;
+		}
+
+		/* MY LIST (Ma Liste)
+
+		(https://www.rtlplay.be/rtlplay/ma-liste) */
+		case ["ma-liste"].includes(pathname.split("/")[2]): {
+			presenceData.details = (await strings).browsing;
+			presenceData.state = (await strings).viewAPage;
+			if (!privacy) {
+				presenceData.state = (await strings).viewList;
+				if (buttons) {
+					presenceData.buttons = [
+						{
+							label: (await strings).buttonViewPage, // Need to be a general string
+							url: href, // We are not redirecting directly to the raw video stream, it's only the media page
+						},
+					];
+				}
+			}
+			break;
+		}
+
+		/* CATEGORY PAGE / COLLECTION PAGE (Page de catégorie)
+
+		(https://www.rtlplay.be/rtlplay/collection/c2dBY3Rpb24) */
+		case ["collection", "series", "films", "divertissement"].includes(
+			pathname.split("/")[2]
+		): {
+			if (privacy) presenceData.details = (await strings).viewAPage;
+			else {
+				const data = JSON.parse(
+					document.querySelectorAll("script[type='application/ld+json']")[0]
+						.textContent
```

Comment:
If you're getting the first element user querySelector() instead of querySelectorAll()
```suggestion
					document.querySelector("script[type='application/ld+json']")
						.textContent
```